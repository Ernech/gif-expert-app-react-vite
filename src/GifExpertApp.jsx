import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GiffExpertApp = () => {

    const [categories, setCategories] = useState(['dogs'])

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return
        setCategories([newCategory,...categories])

    }


    return (
        <>
            <h1>Gif expert app</h1>
            <AddCategory onNewCategory={value => onAddCategory(value)} />
            {/* <button onClick={onAddCategory}>Agregar</button>  */}
                {
                categories.map(category => 
                 <GifGrid 
                        category={category} 
                        key={category}
                    />
                )}
           
        </>
    )
}