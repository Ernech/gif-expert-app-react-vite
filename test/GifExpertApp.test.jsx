import { render, screen, fireEvent } from "@testing-library/react";
import { GiffExpertApp } from "../src/GifExpertApp"


describe('Test at <GifExpertApp/>.', () => {
    test('should match with snapshot', () => {
       const {container}= render(<GiffExpertApp/>)
        expect(container).toMatchSnapshot()
    });

    test('should add a new category',()=>{
        const inputValue = "Dogs"
        render(<GiffExpertApp/>)
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input, { target:{ value:inputValue } })
        fireEvent.submit(form);
        expect(screen.getAllByRole('heading',{level:3}).length).toBe(2)
    
    })

    test('should not add a new category if already exists', () => {
        render(<GiffExpertApp/>)
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input, { target:{ value:"Crash Bandicoot" } })
        fireEvent.submit(form);
        fireEvent.input(input, { target:{ value:"Megadeth" } })
        fireEvent.submit(form);
        fireEvent.input(input, { target:{ value:"Crash Bandicoot" } })
        fireEvent.submit(form);
        expect(screen.getAllByRole('heading',{level:3}).length).toBe(3)
    
    });
});