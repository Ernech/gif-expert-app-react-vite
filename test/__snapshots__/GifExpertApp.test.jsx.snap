// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at <GifExpertApp/>. should match with snapshot 1`] = `
<div>
  <h1>
    Gif expert app
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    dogs
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
